tests/overrides/test_data_parallel.py

```diff
@@ -57,18 +57,25 @@ def test_lightning_wrapper_module_warn_none_output(wrapper_class):
     pl_module.training_step.return_value = None
     pl_module.validation_step.return_value = None
     pl_module.test_step.return_value = None
+    pl_module.predict.return_value = None
 
     with pytest.warns(UserWarning, match="Your training_step returned None"):
         pl_module.running_stage = RunningStage.TRAINING
         wrapped_module()
 
-    with pytest.warns(UserWarning, match="Your test_step returned None"):
+    with pytest.warns(None, match="Your test_step returned None") as record:
         pl_module.running_stage = RunningStage.TESTING
         wrapped_module()
+        assert not record
 
-    with pytest.warns(UserWarning, match="Your validation_step returned None"):
+    with pytest.warns(None, match="Your validation_step returned None") as record:
         pl_module.running_stage = RunningStage.EVALUATING
         wrapped_module()
+        assert not record
+
+    with pytest.warns(UserWarning, match="Your predict returned None"):
+        pl_module.running_stage = RunningStage.PREDICTING
+        wrapped_module()
```

Comment:
I believe predict can't return `None` right?? I haven't tried it out yet.

Reply:
also, there is no warning when predict doesn't return anything in non-distributed env. should I add that?

Reply:
it should be fine not to warn because in contrast to the training step there are no undesired side effects afaik. The reason for warning in the case of training_step is so that the user is aware they are actually skipping steps and not do optimization

Reply:
hmm. I see it isn't doing anything special.. just appending the predictions... no need :+1: